src/NuGet.Core/NuGet.Protocol/HttpSource/HttpRetryHandler.cs

```diff
@@ -90,7 +90,7 @@ internal HttpRetryHandler(IEnvironmentVariableReader environmentVariableReader)
                         {
                             log.LogVerbose("Enhanced retry: HttpRetryHandler is in a state that retry would have been abandoned or not waited if it were not enabled.");
                         }
-                        await Task.Delay(TimeSpan.FromMilliseconds((Math.Pow(2, tries) * request.RetryDelay.TotalMilliseconds) + new Random().Next(200)));
+                        await Task.Delay(TimeSpan.FromMilliseconds((Math.Pow(2, tries) * request.RetryDelay.TotalMilliseconds) + new Random().Next(200)), cancellationToken);
```

Comment:
Our unit tests found a bug when this became the default, this `Task.Delay()` was missing cancellation support.